tasks/hyrax-metadata-updates/index.js

```diff
@@ -271,30 +290,31 @@ async function updateSingleGranule(config, granuleObject) {
     await validate('granule', updatedMetadata, metadataFile.name, config.cmr.provider);
   }
   // Write back out to S3 in the same location
-  await s3PutObject({
-    Bucket: Bucket,
-    Key: Key,
+  const { ETag: newEtag } = await s3PutObject({
+    Bucket,
+    Key,
     Body: updatedMetadata,
     ContentType: metadataResult.ContentType,
     Tagging: s3TagSetToQueryString(tags.TagSet)
   });
-}
+
+  return {
+    ...granule,
+    files: granule.files.map(
+      (file) => (file === metadataFile ? assoc('etag', newEtag, file) : file)
+    )
+  };
+};
 
 /**
  * Update the metadata of each granule with an OPeNDAP data acquisition URL.
  *
  * @param {Object} event - input from the message adapter
  * @returns {Object} the granules
  */
-async function hyraxMetadataUpdate(event) {
-  const granulesInput = event.input.granules;
-
-  await Promise.all(
-    granulesInput.map((granuleObject) => updateSingleGranule(event.config, granuleObject))
-  );
-  // We don't create anything, we just update existing metadata. So we return what we got.
```

Comment:
One: refactoring this function wasn't really necessary. But more importantly, I think the code comment was perhaps useful context, so even if we were going to refactor, I wouldn't have removed it

Reply:
The refactoring was due to my adding the `etag` property to the metadata file of each input granule, and as part of that I avoided using mutation in the `updateSingleGranule` function, which meant changing that function to return the cloned/updated granule object instead of not returning anything at all.

Before, since no changes were made to the granule object (and the function didn't return anything) the comment made sense.  It no longer makes sense because now we _do_ create something (the `etag` properties).

Reply:
Fair enough, that makes sense 👍